reconcile/blackbox_exporter_endpoint_monitoring.py

```diff
@@ -27,6 +27,8 @@ def run(
 ) -> None:
     # verify that only allowed blackbox-exporter modules are used
     settings = queries.get_app_interface_settings()
+    if not settings:
```

Comment:
nit: could we change `get_app_interface_settings()` to have a kwarg `raise_if_empty=False` and set it to true for the cases that we want an exception to be raised? Also, would there be any value in changing the exception to be called something like `AppInterfaceEmptySettingsError` in case it's helpful to handle that different than a general `Exception`?

Reply:
I'll refactor the raise of the exception into `queries.get_app_interface_settings` and `queries.get_app_interface_settings_class`. this makes the code much clearer on the usage site. And yes I'll create a separate exception class for it